src/storage/src/table/cell_based_table.rs

```diff
@@ -49,33 +47,38 @@ pub struct CellBasedTable<S: StateStore> {
     /// The keyspace that the pk and value of the original table has.
     keyspace: Keyspace<S>,
 
-    /// The schema of this table viewed by some source executor, e.g. RowSeqScanExecutor.
-    schema: Schema,
+    /// All columns of this table. Note that this is different from the output columns in
+    /// `mapping.output_columns`.
+    #[allow(dead_code)]
+    table_columns: Vec<ColumnDesc>,
 
-    /// `ColumnDesc` contains strictly more info than `schema`.
-    column_descs: Vec<ColumnDesc>,
+    /// The schema of the output columns, i.e., this table VIEWED BY some executor like
+    /// RowSeqScanExecutor.
+    schema: Schema,
 
-    /// Mapping from column id to column index
+    /// Used for serializing the primary key.
     pk_serializer: OrderedRowSerializer,
 
     /// Used for serializing the row.
     cell_based_row_serializer: CellBasedRowSerializer,
 
-    /// Used for deserializing the row.
+    /// Mapping from column id to column index. Used for deserializing the row.
     mapping: Arc<ColumnDescMapping>,
 
-    column_ids: Vec<ColumnId>,
+    /// Indices of primary keys.
+    /// Note that the index is based on the all columns of the table, instead of the output ones.
+    // FIXME: revisit constructions and usages.
+    pk_indices: Vec<usize>,
 
-    /// Indices of distribution keys in full row for computing value meta. None if value meta is
-    /// not required.
+    /// Indices of distribution keys for computing value meta. None if value meta is not required.
+    /// Note that the index is based on the all columns of the table, instead of the output ones.
+    // FIXME: revisit constructions and usages.
     dist_key_indices: Option<Vec<usize>>,
```

Comment:
Which is better, `Option<Vec<usize>>` or `Vec<usize>`? 🤔

Reply:
Good catch. I think both are acceptable.